content/md/en/docs/reference/command-line-tools/node-template.md

```diff
@@ -147,86 +147,125 @@ Use the `node-template benchmark` command to analyze the resources required to e
 | `-s`, `--steps <steps>` | Specifies how many samples to take across the variable components. The default is one sample.
 | `--template <template>` | Specifies the path to a Handlebars template file used for outputting benchmark results.
-| `--tracing-receiver <receiver>` | Specifies the receiver to process tracing messages. The default is Log.
-| `--tracing-targets <targets>` | Sets a custom profiling filter. Syntax is the same as for logging: `<target>=<level>`.
-| `--wasm-execution <method>` | Specifies the method for executing Wasm runtime code. Valid values are `interpreted`, or `compiled`. The default is `Compiled`.
 
-#### Examples
+For examples of different benchmarking subcommands and the related command-line options, see [Benchmarking examples](#benchmarking-examples).
+
+#### Benchmarking examples
 
-To start a node and benchmark all of the extrinsic calls in all of the pallets, you might a command similar to the following:
+After you have compiled the runtime with benchmarking enabled, you can run a command similar to the following to benchmark all of the function calls in all of the pallets that have runtime-benchmarking configured:
 
 ```shell
-./target/release/node-template benchmark \
+./target/release/node-template benchmark pallet \
     --chain dev \
-    --execution wasm \
-    --wasm-execution compiled \
-    --extrinsic '\*' \
-    --pallet '\*' \
-    --output ./pallets/example/weights.rs
+    --extrinsic "*" \
+    --pallet "*" \
+    --output pallets/weights.rs
+```
+
+With this command, each function call is executed once with a single value and the resulting weight is recorded in the `weights.rs` file.
+
+Depending on the function you want to benchmark, you can add the `--steps` and `--repeat` command-line options to execute the call multiple times with different values.
+For example, the following command executes the `do_something` function in the `pallet_template` and calls the function 20 times to take 10 data points:
+
+```shell
+./target/release/node-template benchmark pallet \
+    --chain dev \
+    --extrinsic do_something \
+    --pallet pallet_template \
+    --steps 10 \
+    --repeat 20 \
+    --output pallets/weight2.rs
+```
+
+To measure the average, median, minimum, and maximum execution time per-block and per-extrinsic, you can run the `node-template benchmark overhead` subcommand:
+
+```shell
+./target/release/node-template benchmark overhead
+```
+
+By default, the command executes the benchmark 100 times, generates results, and writes the output to the `block_weights.rs` and `extrinsics_weights.rs` files.
+You can use command-line options to adjust the calculated weight by adding units or by multiplying the average execution time by some factor.
+
+To measure the storage execution time for the Substrate development chain specification, you can run the following command:
+
+```shell
+./target/release/node-template benchmark storage --state-version 1
 ```
```

Comment:
Same here with the output:  
```pre
Warmup round 1/1    
Preparing keys from block BlockId::Number(0)    
Reading 36 keys    
Time summary [ns]:
	Total: 478367
	Min: 8889, Max: 32258
	Average: 13287, Median: 12580, Stddev: 4189.73
	Percentiles 99th, 95th, 75th: 32258, 21863, 14563
Value size summary:
	Total: 157217
	Min: 1, Max: 155944
	Average: 4367, Median: 16, Stddev: 25621.19
	Percentiles 99th, 95th, 75th: 155944, 82, 80    
Warmup round 1/1    
Preparing keys from block BlockId::Number(0)    
Writing 36 keys    
Time summary [ns]:
	Total: 1877401
	Min: 13476, Max: 665226
	Average: 52150, Median: 25740, Stddev: 111805.24
	Percentiles 99th, 95th, 75th: 665226, 252697, 30427
Value size summary:
	Total: 157217
	Min: 1, Max: 155944
	Average: 4367, Median: 16, Stddev: 25621.19
	Percentiles 99th, 95th, 75th: 155944, 82, 80    
Writing weights to "rocksdb_weights.rs"
```

The two important arguments for this one are `--db paritydb` if you want to use a ParityDB snapshot instead.  
And to use `--state-version 0` for real chain snapshots, as the `--help` indicates. You can also link the [README](https://github.com/paritytech/substrate/tree/master/utils/frame/benchmarking-cli/src/storage#the-benchmark-storage-command) which explains it a bit.